pkg/controller/utils/controllerutil.go

```diff
@@ -34,7 +34,17 @@ func PopulateCrPermissionClusterRoleNames(subjectPermission *managedv1alpha1.Sub
 	return permissionClusterRoleNames
 }
 
-// AllowedNamespaceList 1st pass - allowedRegex
+// GenerateSafeList by 1st checking allow regex then check denied regex
+func GenerateSafeList(allowedRegex string, deniedRegex string, nsList *corev1.NamespaceList) []string {
+
+	safeList := AllowedNamespacesList(allowedRegex, nsList)
+	updatedSafeList := SafeListAfterDeniedRegex(deniedRegex, safeList)
+
+	return updatedSafeList
+
+}
+
+// AllowedNamespacesList 1st pass - allowedRegex
 func AllowedNamespacesList(allowedRegex string, nsList *corev1.NamespaceList) []string {
```

Comment:
I would consider making this and `SafeListAfterDeniedRegex` unexported so they can only be used in `GenerateSafeList`

Reply:
Will do. I was using this function in the test file for subjectPermission controller. But the tests can be changed.